tests/Unit/Evolution/DgSubcell/Test_GhostZoneLogicalCoordinates.cpp

```diff
@@ -0,0 +1,85 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/Slice.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Evolution/DgSubcell/GhostZoneLogicalCoordinates.hpp"
+#include "Evolution/DgSubcell/Mesh.hpp"
+#include "Evolution/DgSubcell/SliceTensor.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace evolution::dg::subcell::fd {
+namespace {
+template <size_t Dim>
+void test() {
+  // Create a subcell mesh
+  const Mesh<Dim> dg_mesh{3, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+  const Mesh<Dim> subcell_mesh = mesh(dg_mesh);
+  const Index<Dim> subcell_extents = subcell_mesh.extents();
+
+  for (const auto& direction : Direction<Dim>::all_directions()) {
+    const size_t dim_direction = direction.dimension();
+
+    // iterate ghost zone size up to the maximum possible value (mesh extent to
+    // the direction)
+    for (size_t ghost_zone_size = 1;
+         ghost_zone_size <= subcell_extents[dim_direction]; ++ghost_zone_size) {
+      const tnsr::I<DataVector, Dim, Frame::ElementLogical> ghost_zone_coords =
+          ghost_zone_logical_coordinates(subcell_mesh, ghost_zone_size,
+                                         direction);
+
+      // grid extents of ghost zone
+      Index<Dim> ghost_zone_extents{subcell_extents};
+      ghost_zone_extents[dim_direction] = ghost_zone_size;
+
+      // Now check the computed ghost zone coords slice-by-slice.
+
+      // First we get the outermost slice of subcell (volume) logical
+      // coordinates to copy the coordinate components that need to remain same.
+      // i.e. For 3D case, if `direction` is along x-axis, then y and z
+      // components of coordinates would remain same as volume slice. (*)
+      auto expected_coordinates = slice_tensor_for_subcell(
+          logical_coordinates(subcell_mesh), subcell_extents, 1, direction);
+
+      for (size_t i_slice = 0; i_slice < ghost_zone_size; ++i_slice) {
+        auto ghost_zone_coords_ith_slice = data_on_slice(
+            ghost_zone_coords, ghost_zone_extents, dim_direction, i_slice);
+
+        // (*) Here we tweak the components that need to be shifted.
+        for (size_t dim = 0; dim < Dim; ++dim) {
```

Comment:
Sorry, why loop over `Dim` if you only check `dim == dim_direction`? Are you missing an `else` branch?

Reply:
Seems like a relic of old struggle.. deleted it